meshcapade_addon/operators.py

```diff
@@ -0,0 +1,1603 @@
+                    image = bpy.data.images[texture_name]
+                else:
+                    if texture_name not in bpy.data.images:
+                        texture_path = os.path.join(PATH, "data", texture_name)
+                        image = bpy.data.images.load(texture_path)
+                    else:
+                        image = bpy.data.images[texture_name]
+
+                node_texture.image = image
+
+                # Link texture node to shader node if not already linked
+                if len(node_texture.outputs[0].links) == 0:
+                    node_tree.links.new(node_texture.outputs[0], node_shader.inputs[0])
+
+        # Switch viewport shading to Material Preview to show texture
+        if bpy.context.space_data:
+            if bpy.context.space_data.type == 'VIEW_3D':
+                bpy.context.space_data.shading.type = 'MATERIAL'
+
+        return {'FINISHED'}
+
+
+class OP_MeasurementsToShape(bpy.types.Operator):
+    bl_idname = "object.measurements_to_shape"
+    bl_label = "Measurements To Shape"
+    bl_description = ("Calculate and set shape parameters for specified measurements")
+    bl_options = {'REGISTER', 'UNDO'}
+
+    betas_regressor_female = None
+    betas_regressor_male = None
+    betas_regressor_neutral = None
+
+    @classmethod
+    def poll(cls, context):
+        try:
+            # Enable button only if mesh is active object
+            return ((context.object.type == 'MESH') and (context.object.parent.type == 'ARMATURE'))
+        except: return False
+
+    def execute(self, context):
+        obj = bpy.context.object
+        bpy.ops.object.mode_set(mode='OBJECT')
+
+        if self.betas_regressor_female is None:
+            regressor_path = os.path.join(PATH, "data", "measurements_to_betas_female.json")
+            with open(regressor_path) as f:
+                data = json.load(f)
+                self.betas_regressor_female = (
+                    np.asarray(data["A"]).reshape(-1, 2), 
+                    np.asarray(data["B"]).reshape(-1, 1)
+                )
+
+        if self.betas_regressor_male is None:
+            regressor_path = os.path.join(PATH, "data", "measurements_to_betas_male.json")
+            with open(regressor_path) as f:
+                data = json.load(f)
+                self.betas_regressor_male = (
+                    np.asarray(data["A"]).reshape(-1, 2),
+                    np.asarray(data["B"]).reshape(-1, 1)
+                )
+
+        if self.betas_regressor_neutral is None:
+            regressor_path = os.path.join(PATH, "data", "measurements_to_betas_neutral.json")
+            with open(regressor_path) as f:
+                data = json.load(f)
+                self.betas_regressor_neutral = (
+                    np.asarray(data["A"]).reshape(-1, 2), 
+                    np.asarray(data["B"]).reshape(-1, 1)
+                )
+
+        if "female" in obj.name.lower():
```

Comment:
And here you just parse the gender out of the obj name and access the appropriate entry.